tests/test_osutils.py

```diff
@@ -479,3 +479,23 @@ def test_lazy_unmount(self):
     "snakeoil.osutils._readdir", "snakeoil.osutils", "listdir", "listdir")
 Test_cpy_posix_loaded = mk_cpy_loadable_testcase(
     "snakeoil._posix", "snakeoil.osutils", "normpath", "normpath")
+
+
+@pytest.mark.parametrize(
+    ["size", "binary", "decimal"],
+    [(0, "0.0 B", "0.0 B"),
+     (1, "1.0 B", "1.0 B"),
+     (1000, "1000.0 B", "1.0 kB"),
+     (1024, "1.0 KiB", "1.0 kB"),
+     (1000**2, "976.6 KiB", "1.0 MB"),
+     (1024**2, "1.0 MiB", "1.0 MB"),
+     (1000**3, "953.7 MiB", "1.0 GB"),
+     (1024**3, "1.0 GiB", "1.1 GB"),
+     (1000**8, "847.0 ZiB", "1.0 YB"),
+     (1024**8, "1.0 YiB", "1.2 YB"),
+     (1000**9, "827.2 YiB", "1000.0 YB"),
+     (1024**9, "1024.0 YiB", "1237.9 YB"),
+     ])
+def test_sizeof_fmt(size, binary, decimal):
+    assert [sizeof_fmt(size, binary=x) for x in (True, False)
+            ] == [binary, decimal]
```

Comment:
This is how I originally did it but it meant that if the first test fails, the second one was never done. But I suppose it would be cleaner to move the expected values outside parametrization.